payments/views.py

```diff
@@ -0,0 +1,105 @@
+from django.http import JsonResponse, HttpResponseBadRequest
+from drf_yasg import openapi
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import status
+from rest_framework.views import APIView
+
+from events.events_repository import EventsRepository
+from users.users_service import UsersService
+from .payments_repository import PaymentsRepository
+from .serializers import CreatePaymentSerializer, PaymentURLSerializer
+
+
+class Payments(APIView):
+    events_repository = EventsRepository()
+    users_service = UsersService()
+    payments_repository = PaymentsRepository()
+
+    authorization_token = openapi.Parameter('Authorization', openapi.IN_HEADER,
+                                            description="Authorization token which starts with Bearer",
+                                            type=openapi.TYPE_STRING)
+
+    @swagger_auto_schema(operation_description='Endpoint for confirming payments.',
+                         request_body=CreatePaymentSerializer,
+                         manual_parameters=[authorization_token],
+                         responses={404: None, 202: 'Payment confirmed', 403: 'User is not a participant',
+                                    409: 'User already payed', 405: 'Event does not require payment',
+                                    406: 'Payment not sufficient'},
+
+                         )
+    def post(self, request):
+        jwt = request.headers['Authorization']
+        event_id = request.data['event_id']
+
+        user = self.users_service.fetch_by_jwt(jwt)
+        event = self.events_repository.get_event_by_id(event_id)
+
+        payment = self.payments_repository.get_payment_for_user_and_event(user, event)
+
+        if not event:
+            return JsonResponse(data=None, status=status.HTTP_404_NOT_FOUND, safe=False)
+        if event.price is None:
+            return JsonResponse(data='Event does not require payment', status=status.HTTP_405_METHOD_NOT_ALLOWED,
+                                safe=False)
+        if payment:
+            return JsonResponse(data='User already payed', status=status.HTTP_409_CONFLICT, safe=False)
+        if not event.participants.filter(id=user.id):
+            return JsonResponse(data='User is not a participant', status=status.HTTP_403_FORBIDDEN, safe=False)
```

Comment:
Both codes 400 and 403 are taken already. If I change that to use either It will not be documented properly, because swagger does not allow multiple responses with the same code.